test/extended/dr/common.go

```diff
@@ -649,26 +681,162 @@ EOF
 	return runPodAndWaitForSuccess(oc, pod)
 }
 
-func runPodAndWaitForSuccess(oc *exutil.CLI, pod *applycorev1.PodApplyConfiguration) error {
-	// this is solely to ensure idempotency, especially in case we run it multiple times locally
-	err := wait.Poll(5*time.Second, 5*time.Minute, func() (done bool, err error) {
-		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
-		defer cancel()
+func runDeleteAndRestoreScript(oc *exutil.CLI, restoreNode *corev1.Node, backupNode *corev1.Node, nonRecoveryNodes []*corev1.Node) error {
+	const name = "etcd-restore-pod"
+	const namespace = "openshift-etcd"
+	framework.Logf("running deletes and restore script on node: %v", restoreNode.Name)
 
-		err = oc.AdminKubeClient().CoreV1().Pods(*pod.Namespace).Delete(ctx, *pod.Name, metav1.DeleteOptions{})
-		if err == nil || apierrors.IsNotFound(err) {
-			return true, nil
-		}
-		framework.Logf("pod deletion result [%v]", err)
-		return false, err
-	})
+	backupInternalIp := internalIP(backupNode)
+	restoreInternalIp := internalIP(restoreNode)
+
+	var nonRecoveryIps []string
+	for _, n := range nonRecoveryNodes {
+		nonRecoveryIps = append(nonRecoveryIps, internalIP(n))
+	}
+
+	restoreScript := fmt.Sprintf(`
+		#!/bin/bash
+        set -exuo pipefail
+		
+		# ssh key dance
+		%s
+
+        NODE_IPS=( %s )
+		for i in "${NODE_IPS[@]}"; do
+		  echo "removing etcd static pod on [$i]"
+		  ssh -i $P_KEY -o StrictHostKeyChecking=no -q core@${i} sudo rm -rf /etc/kubernetes/manifests/etcd-pod.yaml 
```

Comment:
I guess this should be enough. Was thinking whether the staticpod controller would re-add the etcd pod manifest but I think that would only happen with a revision rollout.

Reply:
indeed, it does not. We force the revision rollout later in the code